Taarafo.Core.Tests.Unit/Services/Foundations/Groups/GroupServiceTests.Exceptions.Modify.cs

```diff
@@ -0,0 +1,266 @@
+// ---------------------------------------------------------------
+// Copyright (c) Coalition of the Good-Hearted Engineers
+// FREE TO USE TO CONNECT THE WORLD
+// ---------------------------------------------------------------
+
+using System;
+using System.Threading.Tasks;
+using EFxceptions.Models.Exceptions;
+using Microsoft.Data.SqlClient;
+using Microsoft.EntityFrameworkCore;
+using Moq;
+using Taarafo.Core.Models.Groups;
+using Taarafo.Core.Models.Groups.Exceptions;
+using Xunit;
+
+namespace Taarafo.Core.Tests.Unit.Services.Foundations.Groups
+{
+    public partial class GroupServiceTests
+    {
+        [Fact]
+        public async Task ShouldThrowCriticalDependencyExceptionOnUpdateIfSqlErrorOccursAndLogItAsync()
+        {
+            // given
+            Group randomGroup = CreateRandomGroup();
+            SqlException sqlException = GetSqlException();
+
+            var failedGroupStorageException =
+                new FailedGroupStorageException(sqlException);
+
+            var expectedGroupDependencyException =
+                new GroupDependencyException(failedGroupStorageException);
+
+            this.dateTimeBrokerMock.Setup(broker =>
+                broker.GetCurrentDateTimeOffset())
+                    .Throws(sqlException);
+
+            // when
+            ValueTask<Group> modifyGroupTask =
+                this.groupService.UpdateGroupAsync(randomGroup);
+
+            // then
+            await Assert.ThrowsAsync<GroupDependencyException>(() =>
+               modifyGroupTask.AsTask());
+
+            this.dateTimeBrokerMock.Verify(broker =>
+                broker.GetCurrentDateTimeOffset(),
+                    Times.Once);
+
+            this.storageBrokerMock.Verify(broker =>
+                broker.SelectGroupByIdAsync(randomGroup.Id),
+                    Times.Never);
+
+            this.storageBrokerMock.Verify(broker =>
+                broker.UpdateGroupAsync(randomGroup),
+                    Times.Never);
+
+            this.loggingBrokerMock.Verify(broker =>
+                broker.LogCritical(It.Is(SameExceptionAs(
+                    expectedGroupDependencyException))),
+                        Times.Once);
+
+            this.dateTimeBrokerMock.VerifyNoOtherCalls();
+            this.storageBrokerMock.VerifyNoOtherCalls();
+            this.loggingBrokerMock.VerifyNoOtherCalls();
+        }
+
+        [Fact]
+        public async void ShouldThrowValidationExceptionOnUpdateIfReferenceErrorOccursAndLogItAsync()
+        {
+            // given
+            Group someGroup =
```

Comment:
Consider moving all the lines till 81 to a single line code back to back. I do not see much troubling pattern if they are on the same line.